File: boj_unrated/from20230715/1769.cpp
#include <bits/stdc++.h>
using namespace std;

char buf[1000010];
int main() {
    scanf("%s", buf);
    int len = strlen(buf);

    if (len > 1) {
        int x = 0;
        for(int i = 0; i < len; ++i) {
            x += buf[i] - '0';
        }
        int cnt = 1;
        while (x >= 10) {
            int nextx = 0;
            while(x > 0) {
                nextx += x % 10;
                x /= 10;
            }
            x = nextx;
            ++cnt;
        }
        printf("%d\n", cnt);
        if (x % 3 == 0) printf("YES\n");
        else printf("NO\n");
    } else {
        printf("0\n");
        if ((buf[0] - '0') % 3 == 0) printf("YES\n");
        else printf("NO\n");
    }
    
    return 0;
}

File: boj_unrated/from20230715/1769.py
x = int(input())
cnt = 0
while x >= 10 :
    nextx = 0
    while x > 0 :
        nextx = nextx + (x % 10)
        x = x // 10
    x = nextx
    cnt += 1

print(cnt)
if (x % 3 == 0):
    print("YES")
else:
    print("NO")
